provider/openstack/local_test.go

```diff
@@ -1271,7 +1273,30 @@ func (s *localServerSuite) TestEnsureGroup(c *gc.C) {
 	c.Assert(group.Name, gc.Equals, "test group")
 	assertRule(group)
 	id := group.Id
-	// Do it again and check that the existing group is returned.
+
+	// Do it again and check that the existing group is returned
+	// and updated.
+	rules := []neutron.RuleInfoV2{
+		{
+			Direction:    "ingress",
+			IPProtocol:   "tcp",
+			PortRangeMin: 22,
+			PortRangeMax: 22,
+			EthernetType: "IPv4",
+		},
+		{
+			Direction:    "ingress",
+			IPProtocol:   "icmp",
+			EthernetType: "IPv6",
+		},
+	}
+	group, err = openstack.EnsureGroup(s.env, "test group", rules)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(group.Id, gc.Equals, id)
+	c.Assert(group.Name, gc.Equals, "test group")
+	c.Check(len(group.Rules), gc.Equals, 4)
```

Comment:
We need to check the rule contents, not just the number of rules. eg
c.Check(group.Rules. jc.SameContents, expectedRules)